### DISCUSSION TITLE
What is scenario to set `getOnInit` option in `atomWithStorage`?

### DISCUSSION THREAD

**Discussion by xc1427** [pmndrs/jotai#2896] [Q&A] [answered]
I found this option somewhat confusing. Please correct me if I’m mistaken.

The documentation states:  
getOnInit (optional, default: false): A boolean value indicating whether to retrieve the item from storage during initialization. Note that in an SPA, if getOnInit is either not set or set to false, the initial value will always be used instead of the stored value during initialization. To prefer the stored value, set getOnInit to true.

Consider the scenario where the local storage key is `my_locale`. Based on testing with v2.11.0, the behavior is as follows:  
- When `false`: If `my_locale` is not defined, the atom is assigned the `initialValue`. If `my_locale` exists, even its value is `null`, the atom is first assigned `initialValue` then quickly updated to the local storage value after `baseAtom.onMount` is called, provided no exceptions occur during JSON parsing. (This behavior is confusing because, in most cases, the value ultimately comes from local storage.)  
- When `true`: The value is explicitly retrieved from local storage unless an exception occurs during JSON parsing. (This behavior feels more predictable.)

In summary, this option primarily acts as a safeguard for cases where key in local storage is not defined. However, this behavior can be easily handled in user space with no ambiguity without relying on this option.

**Comment by dai-shi** [comment, marked as answer]
In short, Next.js users should use `false`, RN users would use `true`. We default to `false` because many people reported as bugs previously. `false` is required if you use SSR. If `true` works for your case, feel free to use it.
